### ISSUE TITLE
Incorrect mutations at character creation

### ISSUE THREAD

**Issue by TNHT1990** [CleverRaven/Cataclysm-DDA#63908]
### Describe the bug

When you have a mod that allows you to start with mutations, some more advanced mutations/traits (ie. Eater of the Dead) will be replaced by other traits, seem to be prerequisites. (Eater of the Dead turned into Poison Resist and Substance Tolerance or Caprovore). De-selecting these traits produced an error:
DEBUG    : my_traits and my_mutations were out of sync for TOLERANCE

 FUNCTION : void Character::toggle_trait(const trait_id &, const std::string &)
 FILE     : /home/runner/work/Cataclysm-DDA/Cataclysm-DDA/android/app/jni/src/../../../../src/mutation.cpp
 LINE     : 174
 VERSION  : d6ec466

Can't seem to find a way to start the game with that trait so I can proceed to consume ALL THE THINGS!!

### Attach save file

N/A

### Steps to reproduce

Create a world with Extra Mutated Scenarios.
Create a character with one of those (I chose the diminutive one).
Try selecting Eater of the Dead. 
It was replaced by 3 other traits I think. (Poison Resist, substance tolerance, and saprovore)

### Expected behavior

I would expect to gain the correct trait instead of other being chosen and getting locked out of the trait.

### Versions and configuration

Android x64 
0.G d6ec466
Tiles
English
Extra mutation scenarious

**Comment by estebandellasilva** [comment]
Can you check if you have the requirements for that  Mutation? Because maybe your character didnt cross the threshold required to have said mutation 
According to hitchhiker you need chimera or rat post threshold

**Comment by TNHT1990** [comment]
I went in and the same issue does occur with the Diminutive mutation which should be the threshold mutation for mouse. 
Mouse should have access to 'Eater of the Dead', but the same issue occurs when I have Diminutive active. 
I select Eater of the Dead, it is immediately greyed out, I am given two or three of the following traits: Substance Tolerance, Intestinal Fortitude, Saprovore, and Poison Resistant. Interestingly I can't remove Intestinal Fortitude until I remove the others as it's greyed out until then. 
Another similar trait I've found like this is Very Beautiful will select Beautiful instead but Glorious (the mutation above Very Beautiful) works just fine.

**Comment by estebandellasilva** [comment]
Mouse cant have eater of dead only rat or chimera can ... so the point iss that you want a post threshold mutation for a mutationline which you didnt choose

**Comment by TNHT1990** [comment]
I guess the wiki is out of date, still says that Mouse has Eater of the Dead. My apologies. 
Okay... I tried with Survivor for the Rat tree and it... got weird. 
At first I thought it didn't work, Eater of the Dead went grey and Substance Tolerance and Poison Resistant activated. Trying to deactivate them produced the same Debug error about the My_Traits and My_mutations being out of sync. 
However, I was able to now continuously select and de-select Eater of the Dead without any issue. It just went from the light grey to dark grey text instead of green how it should be when active. Going to the final character screen I was able to check and see that it was indeed active despite the font color indicating otherwise. Now all three were active (Eater of the Dead, Poison Resistant, and Substance Tolerance). 
I think I can get what I'm going for now but still feel like some wires are crossed in the logic somewhere. Though likely not a huge issue since these aren't really meant to be selectable from the start.
